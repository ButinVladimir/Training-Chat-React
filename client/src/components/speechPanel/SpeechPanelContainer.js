import { connect } from 'react-redux';
import SpeechPanel from './SpeechPanel';

const mapStateToProps = state => ({
  loginned: state.loginPanel.loginned,
});

const SpeechPanelContainer = connect(mapStateToProps)(SpeechPanel);

export default SpeechPanelContainer;
