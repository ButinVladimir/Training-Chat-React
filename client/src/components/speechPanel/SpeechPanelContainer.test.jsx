import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import SpeechPanelContainer from './SpeechPanelContainer';

describe('SpeechPanelContainer', () => {
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore();
  });

  it('renders without crashing', () => {
    const store = mockStore({
      loginPanel: {
        loginned: true,
      },
      speechPanel: {
        speech: 'speechValue',
        to: [],
      },
    });
    const component = mount(
      <Provider store={store}>
        <SpeechPanelContainer />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
