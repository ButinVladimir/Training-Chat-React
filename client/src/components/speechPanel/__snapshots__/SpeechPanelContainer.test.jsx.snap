// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeechPanelContainer renders without crashing 1`] = `
<div
  className="chat-speech-panel-container"
>
  <div
    className="speech-container"
  >
    <div
      className="speech-input-container"
    >
      <input
        className="speech"
        onChange={[Function]}
        type="text"
        value="speechValue"
      />
      <button
        className="say"
        onClick={[Function]}
        type="button"
      >
        Say
      </button>
    </div>
    <div
      className="speech-to-container"
    />
  </div>
</div>
`;
